pypeit/spectrographs/keck_lris.py

```diff
@@ -1494,7 +1632,11 @@ def lris_read_amp(inp, ext):
         hdu = io.fits_open(inp)
     else:
         hdu = inp
-    n_ext = len(hdu) - 1  # Number of extensions (usually 4)
+    # Count the number of extensions
+    n_ext = 0
+    for kk, ihdu in enumerate(hdu):
+        if 'VidInp' in ihdu.name:
+            n_ext += 1
```

Comment:
`n_ext = np.sum(['VidInp' in h.name for h in hdu])`

Reply:
clean!